python/dataset.cpp

```diff
@@ -635,6 +648,9 @@ void init_dataset(py::module &m) {
         :return: A new variable that contains the union of all masks.
         :rtype: Variable)");
 
+  bind_astype(dataArray);
+  bind_astype(dataProxy);
```

Comment:
Do we need both? `DataArray` is implicitly convertible to `DataProxy`.

Reply:
Yes, as this is adding `astype()` as a member of `DataArray` and `DataProxy`, not just a free function  of `scipp`.

Reply:
:man_facepalming: :+1: